firmware/led-spiout.c

```diff
@@ -192,64 +185,128 @@ void led_set_spi_frequency(uint8_t frequency) {
     }
 }
 
+/* Sets all leds off, without using the interrupt */
+static void _led_all_off_no_interrupt() {
+    /* make sure we disabled the interrupt ! */
+    DISABLE_LED_WRITES;
 
-void led_init() {
+#define WAIT_SPI_TRANSMIT()     do { while (!(SPSR & _BV(SPIF))) ; } while (0)
+    SPDR = 0;
+    for (uint8_t i = 0; i < LED_START_FRAME_BYTES - 1; ++i) {
+        WAIT_SPI_TRANSMIT();
+        SPDR = 0;
+    }
+    for (uint8_t i = 0; i < NUM_LEDS; ++i) {
+        WAIT_SPI_TRANSMIT();
+        SPDR = BRIGHTNESS_MASK | 0;
```

Comment:
I...think we can just say 0 here, no?

Reply:
I think I remember the spec/articles saying the led needs the first 3 bits to one to understand it's a data frame, else it'll understand it as begin/end frame. I think `BRIGHTNESS_MASK` should actually be `LED_SPI_BEGIN_DATA_BITS`...